### PR TITLE
Unit tests for MergeNodesByLocation class

### PR DESCRIPTION
**Pull request by Auriga2**
### Description of the Change
Added unit tests for MergeNodesByLocation class.

I came up with a random number for the Threshold, such that it fulfill the if conditions. Let me know if we need to use a more realistic number.

### Applicable Issues
https://github.com/constellation-app/constellation/issues/413

### CODE REVIEW

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        int vxId1 = graph.addVertex();
+        int vxId2 = graph.addVertex();
+        int vxId3 = graph.addVertex();
+
+        // set the identifier of each vertex to something unique
+        graph.setStringValue(vertexIdentifierAttribute, vxId1, "V1");
+        graph.setStringValue(vertexIdentifierAttribute, vxId2, "V2");
+        graph.setStringValue(vertexIdentifierAttribute, vxId3, "V3");
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:148]: I'd avoid using the raw type

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+        // set the Longitude of each vertex to something unique
+        graph.setStringValue(vertexLongitudeAttribute, vxId1, "35");
+        graph.setStringValue(vertexLongitudeAttribute, vxId2, "45");
+        graph.setStringValue(vertexLongitudeAttribute, vxId3, "56");
+        graph.setStringValue(vertexLongitudeAttribute, vxId4, "46");
+        graph.setStringValue(vertexLongitudeAttribute, vxId5, "57");
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster1 = new HashSet<>();
+        cluster1.addAll(Arrays.asList(new Integer[]{0, 1, 2, 3, 4}));
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:180]: You can specify the elements of `Arrays.asList()` without needing to declare an array as an FYI (i.e. `Arrays.asList(0, 1, 2, 3, 4)` also works)

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        expResult.put(5, cluster1);
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:183]: raw type

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        int threshold = 1600000;
+        boolean selectedOnly = false;
+        MergeNodesByLocation instance = new MergeNodesByLocation();
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster = new HashSet<>();
+        cluster.addAll(Arrays.asList(new Integer[]{0, 1}));
+        expResult.put(5, cluster);
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:204]: raw type

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getNodesToMerge method, of class MergeNodesByLocation, with
+     * selectedOnly = False
+     */
+    @Test
+    public void testGetNodesToMergee_Threshold_1600000_selectedOnlyFalse_2() throws Exception {
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:192]: The impression I get is that this test is the same as the previous one except with different numbers. Generally speaking I would aim to have inputs for your unit tests achieving a particular purpose (e.g. edge cases, trying to get a particular flow of execution, etc)

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+     */
+    @Test
+    public void testGetNodesToMerge_Threshold_10() throws Exception {
+        System.out.println("testGetNodesToMerge_Threshold_10");
+        Comparator<String> leadVertexChooser = null;
+        int threshold = 10;
+        boolean selectedOnly = false;
+        MergeNodesByLocation instance = new MergeNodesByLocation();
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:220]: raw type

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+        graph.setBooleanValue(vertexSelectedAttribute, vxId3, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId4, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId5, true);
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster = new HashSet<>();
+        cluster.addAll(Arrays.asList(new Integer[]{0, 1}));
+        expResult.put(5, cluster);
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:248]: same here

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster1 = new HashSet<>();
+        cluster1.addAll(Arrays.asList(new Integer[]{1, 3}));
+        Set<Integer> cluster2 = new HashSet<>();
+        cluster2.addAll(Arrays.asList(new Integer[]{2, 4}));
+        expResult.put(5, cluster1);
+        expResult.put(6, cluster2);
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:293]: same here

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+        graph.setBooleanValue(vertexSelectedAttribute, vxId3, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId4, false);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId5, true);
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster = new HashSet<>();
+        cluster.addAll(Arrays.asList(new Integer[]{2, 4}));
+        expResult.put(5, cluster);
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java:335]: same here

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by arcturus2 — APPROVED** [review]
Thank you @Auriga2 for this. I've approved it but pending changes suggested by @antares1470 in relation to having raw types.

**Review by antares1470 — APPROVED** [review]
